src/agent/Agent.cc

```diff
@@ -435,8 +435,16 @@ void Agent::handleGetConsoleProcessListPacket(ReadBuffer &packet)
 
     auto processList = std::vector<DWORD>(64);
     auto processCount = GetConsoleProcessList(&processList[0], processList.size());
-    if (processList.size() < processCount) {
-        processList.resize(processCount);
+
+    // The process list can change while we're trying to read it
+    while (processList.size() < processCount) {
+        // Multiplying by two caps the number of iterations
+        const int newSize = processList.size() * 2;
+        if (newSize <= processList.size()) { // Ensure we fail when new size overflows
```

Comment:
If this code were to overflow, `processList.size()` would be 0x40000000, and `newSize` would be -0x80000000 (i.e. `INT_MIN`). `newSize` would be greater than `processList.size()` because it would be converted from int to size_t for the comparison.

I'd prefer to replace "const int newSize = ... ; ... if { ... }" with:
```c++
const auto newSize = std::max<DWORD>(processList.size() * 2, processCount);
```
If you do, also add `#include <algorithm>` to the top.

Overflow shouldn't ever happen. If it somehow does, then there is one code path handling it -- `std::vector::resize` throws `std::bad_alloc`.